src/playdate/system.nim

```diff
@@ -34,9 +37,29 @@ proc getSecondsSinceEpoch* (this: ptr PlaydateSys): tuple[seconds: uint, millise
 type PDCallbackFunction* = proc(): int {.raises: [].}
 var updateCallback: PDCallbackFunction = nil
 
+proc logException*(exception: ref Exception, fatal: bool = false, messagePrefix: string = "") =
+    ## if fatal is true, the game will pause
+    var message = &"{messagePrefix}: {exception.msg}\n{exception.getStackTrace()}"
+
+    for line in message.splitLines():
+        # Log the error to the console line-by-line, total stack trace might be too long for single call
+        playdate.system.logToConsole(line)
+
+    if fatal:
+        playdate.system.error("FATAL:" & exception.msg)
+
+template runCatching*(body: typed, fatal: bool = false, messagePrefix: string = ""): auto =
+    ## if fatal is true, the game will pause
+    try:
+        body()
+    except Exception as e:
+        logException(e, fatal, messagePrefix)
+        when compiles(result): # result is only defined if body is a proc that has a return type
+            result # default value for inferred return type
```

Comment:
default(body) would not work here.
Also, `return` must not be used in a template